### PR TITLE
added e2e tests using playwright framework

### PR DESCRIPTION
**Pull request by nirajacharya2**
## Description
This pull request introduces automated UI testing using Playwright framework. These tests currently runs when using docker:
**_following tests are done as an admin_**
- Registering
- Logging in
- Creating Spaces
- Creating Recipes
- Editing Account Name

### run this tests locally
```
cd tests
npm install
npm run test:e2e e2e/features/*
```

The following test scenarios are added :
- Scenario: superuser tries to sign up with invalid credentials
- Scenario: superuser registers with valid credentials
- Scenario: superuser logs in with valid credentials
- Scenario: superuser logs in with invalid credentials
- Scenario: superuser creates a space
- Scenario: superuser creates a space from space overview page
- Scenario: superuser tries to create space with existing name
- Scenario: superuser creates a new recipe with basic details only
- Scenario: superuser creates a new recipe with steps and ingredients
- Scenario: superuser updates/edits his/her name

### CODE REVIEW

**Comment by vabene1111** [comment]
hi, thanks for your PR. This is quite a bit of code to review and I have not yet worked with frontend testing tools but it looks good overall. 

It will take some time to get this merged as I am super full with work life and some much more pressing tandoor topics but I will try to get this merged at some point.

**Comment by nirajacharya2** [comment]
i can make the PR a bit shorter that only consist of setting up the testing environment. Maybe register and and login tests only.

**Comment by vabene1111** [comment]
thanks  but i think its fine this way, just need to find some time to look at this

File: tests/database_controller/cleanup.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+docker exec -i recipes-db_recipes-1 bash -c 'PGPASSWORD=password psql -h db_recipes -U djangouser -d djangodb -c "truncate auth_user cascade;"'
```
Comment (SagarGi) [inline tests/database_controller/cleanup.sh:2]: In my local machine the database container name is `recipes_db_recipes_1`. but in CI its the same like above. A bit strange.

Comment (SagarGi) [inline tests/database_controller/cleanup.sh:2]: In my local machine the database container name is `recipes_db_recipes_1`. but in CI its the same like above. A bit strange.

**Pull request closed without merging**